### PR TITLE
Room Manage & Transalte

### PR DESCRIPTION
**Pull request by duongmanhhoang**
https://edu-redmine.sun-asterisk.vn/issues/12260

### CODE REVIEW

File: app/Http/Controllers/Admin/RoomController.php [hunk 1]
```diff
@@ -109,7 +122,92 @@ public function edit($location_id, $room_id)
         return view('admin.rooms.edit', $data);
     }
 
-    public function update(Request $request)
+    public function update(UpdateRoomRequest $request,$location_id, $id)
```
Comment (sonnguyen8x) [inline app/Http/Controllers/Admin/RoomController.php]: convention

File: app/Repositories/RoomDetail/RoomDetailRepository.php [hunk 2]
```diff
@@ -27,13 +41,95 @@ public function getDataStore($data)
...
+    }
+
+    public function getLangMap($id)
+    {
+        $room = RoomDetail::find($id);
+        if (is_null($room)) {
             return false;
         }
+        $lang_map = explode(',', $room->lang_map);
+        return $lang_map;
```
Comment (BunnyPi04) [inline app/Repositories/RoomDetail/RoomDetailRepository.php]: thiếu dòng trắng cuối file

File: database/migrations/2019_05_21_144430_create_rooms_table.php [hunk 3]
```diff
@@ -18,8 +18,8 @@ public function up()
             $table->integer('location_id');
             $table->text('image');
             $table->integer('sale_status')->default(0);
-            $table->date('sale_start_at')->nullable();
-            $table->date('sale_end_at')->nullable();
+            $table->string('sale_start_at', 191)->nullable();
+            $table->string('sale_end_at', 191)->nullable();
```
Comment (BunnyPi04) [inline database/migrations/2019_05_21_144430_create_rooms_table.php]: ko sửa migration đã merged

File: resources/views/admin/rooms/index.blade.php [hunk 4]
```diff
@@ -75,9 +76,23 @@ class="image_table"></td>
                                                 <td class="price">{{ $roomDetail->price }} {{ __('messages.currency') }}</td>
                                                 <td>{{ $room->rating }} {{ __('messages.star') }}</td>
                                                 <td>
-                                                    <a href="{{ route('admin.rooms.edit', [$location->id, $room->id]) }}"
+                                                    @if ($roomDetail->lang_parent_id == 0)
+                                                        {{ __('messages.Original') }}
+                                                    @elseif ($roomDetail->lang_parent_id != 0)
+                                                        @php($parent = \App\Models\RoomDetail::find($roomDetail->lang_parent_id))
```
Comment (BunnyPi04) [inline resources/views/admin/rooms/index.blade.php]: ko access model trong view nhé

**Pull request merged**